src/Elastic.Apm/Model/Transaction.cs

```diff
@@ -153,7 +171,7 @@ internal Transaction(ApmAgent agent, string name, string type)
 			}
 
 			// Also mark the sampling decision on the Activity
-			if (IsSampled && _activity != null && !ignoreActivity)
+			if (IsSampled && _activity != null && !ignoreActivity && _createdActivity)
```

Comment:
One scenario which can be relevant here is when we reuse the existing Activity (so `_createdActivity` is `false`), but the recorded flag on that activity is not set. That flag is propagated for downstream services - technically it's not an issue, because we set the `traceparent` flag for the outgoing HTTP based on our `IsSampled` flag, nevertheless I think in that case we still should set the `Recorded` flag for the existing activity.

For doing so, I'm not sure it's enough to just remove the `_createdActivity` check, we should also nullcheck the activity and check if it's id format is `w3c`.

Reply:
> I think in that case we still should set the Recorded flag for the existing activity.

Sounds good to me, I'll update with this change.

> For doing so, I'm not sure it's enough to just remove the _createdActivity check, we should also nullcheck the activity and check if it's id format is w3c.

Removing the check is good enough, as `_activity` is only set to `Activity.Current` when not null and using W3C id format  (and there's a null check for `_activity` already)

Reply:
done 👍